### DISCUSSION TITLE
Enabling feedback for USB microphone hangs

### DISCUSSION THREAD

**Discussion by Kilklang26** [hathach/tinyusb#2506] [General]
### Operating System

Windows 10

### Board

custom STM32F4

### Firmware

Custom

### What happened ?

Hello,
When I add feedback endpoint to my USB microphone application the device enumerates but hangs immediately after.

This is what I did :
- Start from the USB 1 channel microphone example
- Add TUD_AUDIO_DESC_STD_AS_ISO_FB_EP endpoint after the isochronous data endpoint and adapt descriptor size
- Feedback endpoint number is set to the same as data endpoint. Data is an IN endpoint, Feedback is OUT with size=4bytes.
- Set the number of endpoints to 2 in Standard AS Interface Descriptor
 (following the TUD_AUDIO_SPEAKER_MONO_FB_DESCRIPTOR example)

- Enable CFG_TUD_AUDIO_ENABLE_FEEDBACK_EP and CFG_TUD_AUDIO_ENABLE_EP_OUT

Does someone have a working example of UAC2.0 audio input with feedback?

By the way, audio with implicit feedback works well but Windows doesn't support it, I need to implement explicit feedback hence my questions

### How to reproduce ?

--

### Debug Log as txt file (LOG/CFG_TUSB_DEBUG=2)

-

### I have checked existing issues, dicussion and documentation

- [X] I confirm I have checked existing issues, dicussion and documentation.

**Comment by HiFiPhile** [comment]
Feedback is only for output streaming, input streaming doesn't need feedback since the host process exactly the same amount of data you sent and no sample will be discarded.

I suggest you to read USB specification about data flow to have a better understanding.

**Reply by Kilklang26** [reply]
Thanks ! I read so much about USB audio lately, yet I missed that somewhat...
